### PR TITLE
#161255195 Users should be able to login

### PR DESCRIPTION
**Pull request by mmosoroohh**
### What does this PR do?
Creates a login feature that enables registered users to be able to login to authors' haven

### Description of Task to be completed?
- Creates login form
- Creates login reducer.
- Creates action creators to dispatch actions.
- Stores token and user profile in local storage on successful login

### How should this be manually tested?
- Clone this repository.
- Branch out to ft-user-able-login-authors-161255195
- Install node and yarn
- Run ```yarn install```
- Run ```yarn start```
- Navigate to ```Login page```

### What are the relevant pivotal tracker stories?
[#161255195](https://www.pivotaltracker.com/story/show/161255195)

### Screenshots (if appropriate)
- Login page
screen shot 2018-12-05 at 10 12 34

- Login successfully
screen shot 2018-12-05 at 10 07 28

- Login with wrong credentials
screen shot 2018-12-05 at 10 06 34

### Checklist:
- [x] My code follows the style guidelines of this project
- [ ] My teammates have reviewed my PR
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My PR has one commit.###

### CODE REVIEW

File: src/components/SocialButtons/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { Divider } from 'semantic-ui-react';
+
+export const SocialButtons = () => (
```
Comment (Rodgers-M) [inline src/components/SocialButtons/index.js]: you do not have to export here as it is exported at the bottom

File: src/constants/auth.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const TOKEN_KEY = 'jwtToken';
+export const USER_TOKEN = {
```
Comment (Rodgers-M) [inline src/constants/auth.js]: sensitive information should be stored in environment variables

File: src/containers/Login/index.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      email: '',
+      password: '',
+      errors: {},
+    };
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  onChange = (e) => {
+    // console.log('changed', e.target.name);
```
Comment (Rodgers-M) [inline src/containers/Login/index.js]: please remove all log statements

File: src/containers/Login/index.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        value: password,
+        id: 'password',
+        type: 'password',
+        icon: 'lock',
+        placeholder: 'Enter password',
+        onChange: this.onChange,
+      },
+    ];
+    return (
+      <div className="ui raised very padded center aligned text container segment container-main">
```
Comment (Rodgers-M) [inline src/containers/Login/index.js]: i think this `jsx` can be abstracted to a method, just like what you have done to `renderMessage` or create a component and pass the necessary props to it then render the component here

File: src/containers/Login/index.js (hunk 3 above)
Comment (mmosoroohh) [reply src/containers/Login/index.js]: okay. Ill work on it asap

File: src/components/SocialButtons/index.js (hunk 1 above)
Comment (mmosoroohh) [reply src/components/SocialButtons/index.js]: okay

File: src/redux/actions/__tests__/login.actions.test.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  });
+});
+
+describe('Login error action creators', () => {
+  it('should dispatch LOGIN_ERROR type', () => {
+    expect(loginError({}).type).toEqual(LOGIN_ERROR);
+  });
+});
+
+describe('Login success action creator', () => {
```
[the remark this answers is not in the document]
Comment (Rodgers-M) [reply src/redux/actions/__tests__/login.actions.test.js:20]: @mmosoroohh you can disable this by excluding patterns in tests , have a look at this 
[code climate resource](https://docs.codeclimate.com/docs/excluding-files-and-folders)

**Review by Rodgers-M — CHANGES_REQUESTED** [review]
have a look at the comments. then on exporting files, only redux connected components can be exported twice for testing purposes

**Review by Rodgers-M — APPROVED** [review]

**Pull request merged**